temporalio/lib/temporalio/runtime.rb

```diff
@@ -105,23 +125,28 @@ def _to_bridge
     #   @return [PrometheusMetricsOptions, nil] Prometheus options if using Prometheus. This is mutually exclusive with
     #     +opentelemetry+.
     # @!attribute attach_service_name
-    #   @return [Boolean] Whether to put the service_name on every metric, default +true+.
+    #   @return [Boolean] Whether to put the service_name on every metric.
     # @!attribute global_tags
     #   @return [Hash<String, String>, nil] Resource tags to be applied to all metrics.
     # @!attribute metric_prefix
-    #   @return [String, nil] Prefix to put on every Temporal metric. If unset, defaults to +temporal_+.
-    MetricsOptions = Struct.new(
-      :opentelemetry,
-      :prometheus,
-      :attach_service_name,
-      :global_tags,
-      :metric_prefix,
-      keyword_init: true
-    ) do
-      # @!visibility private
-      def initialize(**kwargs)
-        # @type var kwargs: untyped
-        kwargs[:attach_service_name] = true unless kwargs.key?(:attach_service_name)
+    #   @return [String, nil] Prefix to put on every Temporal metric. If unset, defaults to `temporal_`.
+    class MetricsOptions
+      # Create metrics options. Either `opentelemetry` or `prometheus` required, but not both.
+      #
+      # @param opentelemetry [OpenTelemetryMetricsOptions, nil] OpenTelemetry options if using OpenTelemetry. This is
+      #   mutually exclusive with `prometheus`.
+      # @param prometheus [PrometheusMetricsOptions, nil] Prometheus options if using Prometheus. This is mutually
+      #   exclusive with `opentelemetry`.
```

Comment:
Don't really love that. Can we just have two methods?

Reply:
This is a constructor and Ruby does not have overloads so you don't want two options. One thing we can do is have one union option. This is what we do in Python, but in .NET for typing reasons we have separate mutually exclusive fields. Granted Python brought these "metrics" specific options up to the Telemetry layer which we shouldn't do again, so they got to call this field `metrics: Optional[Union[OpenTelemetryConfig, PrometheusConfig, MetricBuffer]]`. If we combine for Ruby, what should we call the field? Surely not "metrics". Otherwise, I think the mutually exclusive options like .NET is fine.

Reply:
Static helpers would be OK by me I think, but, if that's atypical we don't have to

Reply:
I think it's a bit atypical. Having mutually exclusive kwargs is pretty common. `MetricsOptions.new(opentelemetry: whatever)` or `MetricsOptions.new(prometheus: whatever)` reads well.